icu4c/source/test/intltest/loctest.cpp

```diff
@@ -4514,6 +4515,35 @@ LocaleTest::TestGetUnicodeKeywordValueStdString() {
     assertEquals(keyword, expected, result.c_str());
 }
 
+void
+LocaleTest::TestSetKeywordValueImp() {
+    IcuTestErrorCode status(*this, "TestSetKeywordValueImp()");
+
+    {
+        CharString localeID("aa", status);
+        ulocimp_setKeywordValue("bb", "cc", localeID, status);
+        assertEquals("", "aa@bb=cc", localeID.data());
```

Comment:
ufield(Also known as keyword) | = ukey (sep uvalue)?
-- | --

and



ukey(Also known as key) | = alphanum alpha ;(Note that this is narrower than in [RFC6067], so that it is disjoint with tkey.)
-- | --


 
and


uvalue(Also known as type) | = alphanum{3,8}(sep alphanum{3,8})* ;
-- | --

so cc is not a valid uvalue

and in theory we could have zz as ukey and zz could still be a ufield right (since  (sep uvalue)? )